### DISCUSSION TITLE
torbrowser-launcher: does not launch with dark theme

### DISCUSSION THREAD

**Discussion by Crehet** [netblue30/firejail#5843] [Q&A] [answered]
launching firejailed [torbrowser-launcher](https://archlinux.org/packages/extra/any/torbrowser-launcher/) with its own firejail profile in `/etc/firejail` doesn't start with a dark theme, it uses light theme.
even though `whitelist-common.profile` already gives it permission to read the gtk themes.

**Comment by ghost** [comment]
Good question. IMO there's nothing wrong with your expectation or assumption in this context. From experience though I've noticed that this is more like a suggestion - like in 'would you consider following this setting' - than a strictly enforced choice. Some applications do and others don't, regardless of using Firejail.

There are two places where Firefox-based browsers show GUI settings for following the OS (or 'system') theme. One is on the `about:preferences` page > Language and Appearance > Website appearance. The other is on the `about:addons` page > Themes. FWIW, when I change the value of `gtk-application-prefer-dark-theme` from 0 to 1 in ${HOME}/.config/gtk-3.0/settings.ini I never see it respected by either Firefox or TorBrowser (when using their internal 'system (auto)' prefs to follow OS settings.

Do you actually notice any difference between running TB with versus without firejail?

**Reply by Crehet** [reply]
true. neither Firefox nor torbrowser-launcher respect settings in `${HOME}/.config/gtk-3.0/settings.ini`

>Do you actually notice any difference between running TB with versus without firejail?

without firejail it does use dark theme. and with firejail it uses light theme.

when i use `firejail --noprofile  --dbus-user=none torbrowser-launcher` it starts with light theme and if i remove `--dbus-user=none` it starts with dark theme

**Reply by ghost** [reply]
> when i use firejail --noprofile  --dbus-user=none torbrowser-launcher it starts with light theme and if i remove --dbus-user=none it starts with dark theme

You _could_ add `ignore dbus-user none` in a torbrowser-launcher.local to have TB start with a dark theme. Personally I'd prefer to keep its sandbox as tight as possible and simply set a dark theme via TB's GUI. Enabling unfiltered access to the D-Bus user interface is usually advised against. The decision is yours of course, just be aware of the potential security risks involved. You might try to determine what specific user D-Bus call is responsible for theming and use `dbus-user filter` instead of fully ignoring dbus-user.

**Reply by rusty-snake** [reply]
`org.freedesktop.portal.Desktop` I guess. But you need to drop `noroot`.

**Reply by Crehet** [reply, marked as answer]
> `org.freedesktop.portal.Desktop` I guess. But you need to drop `noroot`.

didn't need to drop `noroot`.

but it worked. `firejail --noprofile --env=GDK_BACKEND=wayland --dbus-system=none --dbus-user=filter --dbus-user.talk=org.freedesktop.portal.Desktop --noroot torbrowser-launcher`
